woocommerce/woocommerce-ios
# Readonly Product Variations: Storage layer schema update

Fixes #1467 

## Changes
- Discussion: p99K0U-1Za-p2
- 530 diffs are on the Core Data model
- Added the following entities to a new model version 23:
  - `Attribute`: meant to be a generic attribute with 3 core properties: `id`, `key`, `value`
  - `ProductVariation`: following the attributes as defined on the WC API doc in the table below
    - READ-ONLY attributes are non-optional
    - Plus `productID` and `siteID` attributes so that we can look up the variations given a site ID and product ID later

## Testing
- Launch the app from `develop`, logged in
- Launch the app from the PR branch --> the app should run normally and not crash

API attribute | type | description
-- | -- | --
id | integer | Unique identifier for the resource.READ-ONLY
date_created | date-time | The date the variation was created, in the site's timezone.READ-ONLY
date_created_gmt | date-time | The date the variation was created, as GMT.READ-ONLY
date_modified | date-time | The date the variation was last modified, in the site's timezone.READ-ONLY
date_modified_gmt | date-time | The date the variation was last modified, as GMT.READ-ONLY
description | string | Variation description.
permalink | string | Variation URL.READ-ONLY
sku | string | Unique identifier.
price | string | Current variation price.READ-ONLY
regular_price | string | Variation regular price.
sale_price | string | Variation sale price.
date_on_sale_from | date-time | Start date of sale price, in the site's timezone.
date_on_sale_from_gmt | date-time | Start date of sale price, as GMT.
date_on_sale_to | date-time | End date of sale price, in the site's timezone.
date_on_sale_to_gmt | date-time | End date of sale price, as GMT.
on_sale | boolean | Shows if the variation is on sale.READ-ONLY
status | string | Variation status. Options:draft,pending,privateandpublish. Default ispublish.
purchasable | boolean | Shows if the variation can be bought.READ-ONLY
virtual | boolean | If the variation is virtual. Default isfalse.
downloadable | boolean | If the variation is downloadable. Default isfalse.
downloads | array | List of downloadable files. SeeProduct variation - Downloads properties
download_limit | integer | Number of times downloadable files can be downloaded after purchase. Default is-1.
download_expiry | integer | Number of days until access to downloadable files expires. Default is-1.
tax_status | string | Tax status. Options:taxable,shippingandnone. Default istaxable.
tax_class | string | Tax class.
manage_stock | boolean | Stock management at variation level. Default isfalse.
stock_quantity | integer | Stock quantity.
stock_status | string | Controls the stock status of the product. Options:instock,outofstock,onbackorder. Default isinstock.
backorders | string | If managing stock, this controls if backorders are allowed. Options:no,notifyandyes. Default isno.
backorders_allowed | boolean | Shows if backorders are allowed.READ-ONLY
backordered | boolean | Shows if the variation is on backordered.READ-ONLY
weight | string | Variation weight.
dimensions | object | Variation dimensions. SeeProduct variation - Dimensions properties
shipping_class | string | Shipping class slug.
shipping_class_id | string | Shipping class ID.READ-ONLY
image | object | Variation image data. SeeProduct variation - Image properties
attributes | array | List of attributes. SeeProduct variation - Attributes properties
menu_order | integer | Menu order, used to custom sort products.

Update release notes:
